src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -176,8 +176,21 @@ namespace OpenMS
     connect(&tab_bar_, &EnhancedTabBar::dropOnTab, this, &TOPPViewBase::copyLayer);
     box_layout->addWidget(&tab_bar_);
 
+    //TODO This triggers always also if there are just focus changes. Very very bad..
     connect(&ws_, &EnhancedWorkspace::subWindowActivated, [this](QMdiSubWindow* window) {
-      if (window != nullptr) /* 0 upon terminate */ updateBarsAndMenus(); 
+      Size s = 0;
+      int oldactiveSubwindowID = activeSubwindowID_;
+      for (const auto& child : ws_.subWindowList())
```

Comment:
can you add a comment (e.g., only update if id changes? ).
Is the TODO still valid?

Reply:
I think Julianus made some changes and now this does not get triggered in every focus changes

Reply:
No, right, the TODO can go in favor of a comment. I will make a suggestion.